### DISCUSSION TITLE
Listing all files on a single volume

### DISCUSSION THREAD

**Discussion by bradmurray** [seaweedfs/seaweedfs#6273] [General]
Is there an API call to list all file names on a volume? I am trying to clean up the fact that I have a bunch of un-full volumes after my bulk load, but fs.mergeVolumes will not move the larger files so there is no way to get the volumes completely empty without moving those large files out of the file store completely, vacuum the volumes and then move them back.

**Comment by chrislusf** [comment]
This command can list files in a volume https://github.com/seaweedfs/seaweedfs/tree/master/unmaintained/see_dat

**Comment by bradmurray** [comment]
Is there a way to call that via url?

**Reply by chrislusf** [reply]
No. It is "unmaintained".

**Reply by JiangweiYe76** [reply]
@chrislusf Can we pull a request to it, creating a url api to call this. Or is that deprecated?

**Comment by bradmurray** [comment]
I wrote a brute-force bash script to generate a list by iterating the files in my directory and parsing the filer metadata into a text log for each volume that shows the file names in it:
```sh
myweedmount="~/weed"
outputfolder="/tmp"
filerurl="http://weedfiler:9335"
mkdir -f $outputfolder
cd $myweedmount
find . -type f -size +0 | cut -c 3- | while read filerPath; do
	counter=0
	curl -s "$filerurl/$(python -c "import urllib.parse; print (urllib.parse.quote('''$filerPath'''))")?metadata=true" | jq -r ".chunks[].file_id" | cut -d',' -f1 | sort | uniq | while read volumeId; do
		echo "$filerPath" >> $outputfolder/$volumeId.txt"
		counter=1
	done
	if [ $counter -eq 0 ]; then echo "Error on $filerPath"; fi
done
cd -
```

Unfortunately it has to use python to do url encode or the metadata web calls won't work. It also requires [jq](https://jqlang.github.io/jq/)
